libraries/botbuilder-core/src/configurationServiceClientCredentialFactory.ts

```diff
@@ -28,14 +50,87 @@ export type ConfigurationServiceClientCredentialFactoryOptions = z.infer<typeof
  * ServiceClientCredentialsFactory that uses a [ConfigurationServiceClientCredentialFactoryOptions](xref:botbuilder-core.ConfigurationServiceClientCredentialFactoryOptions) or a [Configuration](xref:botbuilder-dialogs-adaptive-runtime-core.Configuration) instance to build ServiceClientCredentials with an AppId and App Password.
  */
 export class ConfigurationServiceClientCredentialFactory extends PasswordServiceClientCredentialFactory {
+    private readonly inner: ServiceClientCredentialsFactory;
+
     /**
      * Initializes a new instance of the [ConfigurationServiceClientCredentialFactory](xref:botbuilder-core.ConfigurationServiceClientCredentialFactory) class.
      *
      * @param factoryOptions A [ConfigurationServiceClientCredentialFactoryOptions](xref:botbuilder-core.ConfigurationServiceClientCredentialFactoryOptions) object.
      */
     constructor(factoryOptions: ConfigurationServiceClientCredentialFactoryOptions = {}) {
-        const { MicrosoftAppId = null, MicrosoftAppPassword = null } = TypedConfig.nonstrict().parse(factoryOptions);
-        super(MicrosoftAppId, MicrosoftAppPassword);
+        // Exclude MicrosoftAppType from Zod to ignore-casing comparison.
+        // .NET code:
+        // https://github.com/microsoft/botbuilder-dotnet/blob/d84c6a1f76a56dbee0d18a16adf5d678e5b30035/libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/ConfigurationServiceClientCredentialFactory.cs#L53-L55
+        const { MicrosoftAppType, ...options } = factoryOptions;
```

Comment:
If we aren't enforcing case-sensitivity, there's no reason to make this an `enum` IMO. Sorry for the bad advice before! Just make the type above a `z.string()` and then compare against a set of `const`s. The point of `zod` is that it provides good error messages by default, and if we aren't going to force users to use the correct casing in the JSON file there's no point.